### PR TITLE
Feature 4218 add disease properties switch between basic and extended classification

### PR DESCRIPTION
**Pull request by sergiupacurariu**

### CODE REVIEW

File: sormas-api/src/main/resources/captions.properties [hunk 1]
```diff
@@ -324,6 +324,7 @@ CaseData.clinicalConfirmation=Clinical confirmation
 CaseData.community=Responsible community
 CaseData.epidemiologicalConfirmation=Epidemiological confirmation
 CaseData.laboratoryDiagnosticConfirmation=Laboratory diagnostic confirmation
+CaseData.confirmedCaseClassification = Confirmed case classification
```
Comment (MateStrysewske) [inline sormas-api/src/main/resources/captions.properties:327]: Call this property `caseConfirmationBasis` and use the caption "Basis for confirmation"

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java [hunk 2]
```diff
@@ -2041,6 +2041,18 @@ public void onUpgrade(SQLiteDatabase db, ConnectionSource connectionSource, int
 				getDao(Sample.class).executeRaw("ALTER TABLE samples ADD COLUMN samplingReason varchar(255);");
 				getDao(Sample.class).executeRaw("ALTER TABLE samples ADD COLUMN samplingReasonDetails text;");
 
+			case 284:
+				currentVersion = 284;
+
+				getDao(DiseaseConfiguration.class).executeRaw("ALTER TABLE diseaseconfiguration ADD COLUMN extendedClassification boolean;");
+				getDao(DiseaseConfiguration.class).executeRaw("ALTER TABLE diseaseconfiguration ADD COLUMN extendedClassificationMulti boolean;");
+				getDao(DiseaseConfiguration.class).executeRaw("UPDATE diseaseconfiguration SET extendedClassification = 0 WHERE disease not in ('CORONAVIRUS', 'MEASLES');");
+				getDao(DiseaseConfiguration.class).executeRaw("UPDATE diseaseconfiguration SET extendedClassificationMulti = 0; WHERE disease not in ('CORONAVIRUS');");
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java:2050]: I should have copy the tested script; I added the Where clause later as it works fine without it but it is better this way. I removed the semicolon.

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/ConfirmedCaseClassification.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package de.symeda.sormas.api.caze;
+
+import de.symeda.sormas.api.i18n.I18nProperties;
+
+public enum ConfirmedCaseClassification {
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-api/src/main/java/de/symeda/sormas/api/caze/ConfirmedCaseClassification.java:5]: changed

File: sormas-app/app/src/main/java/de/symeda/sormas/app/caze/edit/CaseEditFragment.java [hunk 4]
```diff
@@ -203,17 +211,46 @@ private void setUpFieldVisibilities(final FragmentCaseEditLayoutBinding contentB
...
 		}
 
+		updateCaseConfirmationVisibility(contentBinding);
+
 		contentBinding.caseDataQuarantineExtended.setVisibility(record.isQuarantineExtended() ? VISIBLE : GONE);
 		contentBinding.caseDataQuarantineReduced.setVisibility(record.isQuarantineReduced() ? VISIBLE : GONE);
 	}
 
+	private void updateCaseConfirmationVisibility(FragmentCaseEditLayoutBinding contentBinding) {
+		DiseaseConfiguration diseaseConfiguration = DatabaseHelper.getDiseaseConfigurationDao().getDiseaseConfiguration(record.getDisease());
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/caze/edit/CaseEditFragment.java:225]: I overlooked that, changed it.

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**